source/write/update.txt

```diff
@@ -21,15 +21,15 @@ Overview
 --------
 
 In this guide, you can learn how to use the {+driver-short+} to update
-documents in a MongoDB collection by using the ``update_one()`` and
-``update_many()`` methods.
+documents in a MongoDB collection by using the ``update_one`` and
+``update_many`` methods.
```

Comment:
Q: most of our docs include parentheses after a method name; is there something different about ruby?

Reply:
I asked Nora about this – it seems like current Ruby documentation avoids using parentheses. The language in general seems more relaxed about using parentheses in code.

Reply:
Parentheses are largely optional in Ruby, as long as the meaning is unambiguous. By convention, parentheses are almost always omitted when a method call has no parameters (e.g `file.close`). Otherwise, Rubyists often omit parentheses when there is only a single literal argument (e.g. `puts "hello"`) or with keyword arguments (e.g. `foo a: 1, b: "two"`). It becomes a stylistic concern at that point.

Asking MongoGPT about it, the all-knowing AI suggests the following:

In Ruby, omitting parentheses for method arguments is common, especially in certain contexts where it enhances readability and maintains a more natural syntax. Here are some conventions and guidelines that developers often follow when deciding whether to omit parentheses:

1. **DSLs (Domain-Specific Languages):** When designing or using DSLs, parentheses are often omitted to create a syntax that reads more like a natural language. This is common in frameworks like Rails, particularly in routing and validation declarations.

2. **Control Structures:** For built-in Ruby control structures such as `if`, `while`, `until`, and `case`, parentheses are typically omitted around their conditions to improve readability and maintain a cleaner syntax.

3. **Short Methods:** When calling methods that take no arguments or only a single simple argument, it is common to omit parentheses to streamline the code. For example:  
   ```ruby
   puts "Hello, world"
   ```

4. **Internal DSLs or Configure Blocks:** In configurations or setup blocks (e.g., Rake tasks, ActiveRecord scopes), omitting parentheses can make the code easier to follow and more domain-centric.

5. **Chaining Methods:** When method calls are chained, parentheses are usually omitted for the first method call if it takes a single argument, leading to a more fluid syntax.

6. **Readability:** Parentheses can be omitted when they enhance readability by removing unnecessary clutter, especially in expressions that are clearly understood without them.

However, there are also scenarios where including parentheses is preferred:

- **Ambiguity:** If omitting parentheses leads to ambiguity or confusion about the order of operations, it is better to include them.
- **Multi-Argument Methods:** When a method takes multiple arguments, especially with complex expressions, using parentheses can make the argument list clearer.
- **Precedence Issues:** In expressions with complex operator precedence, including parentheses can clarify the intended order of operations.
- **Method Definitions:** When defining methods, it's common to use parentheses for the argument list to clearly specify the parameters, even though it is optional.

Ultimately, the decision to omit parentheses often comes down to personal or team styling preferences, but developers should aim for consistency and clarity in their code.